src/main/java/task/Task.java

```diff
@@ -0,0 +1,232 @@
+package task;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import exception.DukeException;
+import exception.FormatException;
+import exception.NoInputException;
+
+/**
+ * Task class is used to represent a task in Duke.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Abstract method toFileString is used to convert the task to string 
+     * for storage in file with Storage class.
+     * @return The file string representation of the task.
+     */
+    public abstract String toFileString();
+
+    /**
+     * Static method fromFileString is used to convert a string from file
+     * to a task.
+     * @param s The file string representation of the task.
+     * @return The UI version of the task string.
+     */
+    public static Task fromFileString(String s) {
+        String[] parts = s.split("\\|");
+        String type = parts[0];
+        String isDone = parts[1];
+        String description = parts[2];
+
+        Task task;
+        switch (type) {
+            case "T":
```

Comment:
There should be no indentation for "case". Maybe change it to the same indentation as switch keyword?